code/src/components/UserInput.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { game } from "reducers/game";
+import styled from "styled-components/macro";
+
+import { Button } from "lib/Button";
+import { getStartGame } from "../reducers/reusable";
+import { AnimationText } from "lib/AnimationText";
+
+export const UserInput = () => {
+  const [name, setName] = useState("");
+  const username = useSelector(store => store.game.username);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (username) {
+      dispatch(getStartGame());
+    }
+  }, [dispatch, username]);
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    dispatch(
+      game.actions.addUserName({
+        username: name,
+      })
+    );
+    setName(""); // Clearing the input
+  };
+
+  return (
+    <TopSection>
+      <AnimationText />
+      <Form onSubmit={handleSubmit}>
+        <Label>
+          <InputField
+            type="text"
+            placeholder="Enter your username"
+            value={name}
+            onChange={event => setName(event.target.value)}
+          />
+        </Label>
+        <Button buttonDisabled={!name} buttonType="submit" text="Start Game" />
+      </Form>
+    </TopSection>
+  );
+};
+
+const TopSection = styled.section`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  color: #fff;
+  width: 100%;
+  max-width: 530px;
+  @media (min-width: 768px) {
+    max-width: 800px;
+  }
+`;
+
+const Form = styled.form`
+  display: flex;
+  width: 100%;
+  justify-content: space-evenly;
+  align-items: center;
+  margin: 5px 0;
+
+  @media (min-width: 1024px) {
+    margin: 10px 0;
+  }
+`;
+
+const Label = styled.label``;
```

Comment:
Since this is not styled, no need to have this. It could be plain JSX instead. `<label> </label>`